pkg/pillar/scripts/onboot.sh

```diff
@@ -0,0 +1,173 @@
+BINDIR=/opt/zededa/bin
+TMPDIR="${PERSISTDIR}/tmp"
+ZTMPDIR=/run/global
+DPCDIR="$ZTMPDIR/DevicePortConfig"
+FIRSTBOOTFILE="$ZTMPDIR/first-boot"
+FIRSTBOOT=
+TPM_DEVICE_PATH="/dev/tpmrm0"
+SECURITYFSPATH=/sys/kernel/security
+PATH=$BINDIR:$PATH
+DISKSPACE_RECOVERY_LIMIT=70
+
+echo "$(date -Ins -u) Starting onboot.sh"
+
+# Copy pre-defined fscrypt.conf
+cp fscrypt.conf /etc/fscrypt.conf
+
+mkdir -p $ZTMPDIR
+if [ -d $TMPDIR ]; then
+    echo "$(date -Ins -u) Old TMPDIR files:"
+    ls -lt $TMPDIR
+    rm -rf $TMPDIR
+fi
+mkdir -p $TMPDIR
+export TMPDIR
+
+if ! mount -t securityfs securityfs "$SECURITYFSPATH"; then
+    echo "$(date -Ins -u) mounting securityfs failed"
+fi
+
+DIRS="$PERSIST_CERTS $PERSIST_AGENT_DEBUG /persist/status/zedclient/OnboardingStatus"
+
+# If /persist/installer/first-boot exists treat this as a first boot
+# we rename file to not assume that it is the first boot if we reboot occasionally
+if [ -f "$PERSISTDIR/installer/first-boot" ]; then
+    mv "$PERSISTDIR/installer/first-boot" "$PERSISTDIR/installer/send-require"
+    touch $FIRSTBOOTFILE # For nodeagent
+    FIRSTBOOT=1
+fi
+
+# If /persist didn't exist or was removed treat this as a first boot
+if [ ! -d $PERSIST_CERTS ]; then
+    touch $FIRSTBOOTFILE # For nodeagent
+    FIRSTBOOT=1
+fi
+
+for d in $DIRS; do
+    d1=$(dirname "$d")
+    if [ ! -d "$d1" ]; then
+        echo "$(date -Ins -u) Create $d1"
+        mkdir -p "$d1"
+        chmod 700 "$d1"
+    fi
+    if [ ! -d "$d" ]; then
+        echo "$(date -Ins -u) Create $d"
+        mkdir -p "$d"
+        chmod 700 "$d"
+    fi
+done
+
+# Save any existing checkpoint directory for debugging
+rm -rf $PERSIST_AGENT_DEBUG/checkpoint
+if [ -d /persist/checkpoint ]; then
+    echo "$(date -Ins -u) Saving copy of /persist/checkpoint in /persist/agentdebug"
+    cp -rp /persist/checkpoint $PERSIST_AGENT_DEBUG/
+fi
+
+# Save any existing /persist/status directory for debugging
+rm -rf $PERSIST_AGENT_DEBUG/status
+if [ -d /persist/status ]; then
+    echo "$(date -Ins -u) Saving copy of /persist/status in /persist/agentdebug"
+    cp -rp /persist/status $PERSIST_AGENT_DEBUG/
+fi
+
+echo "$(date -Ins -u) Configuration from factory/install:"
+(cd $CONFIGDIR || return; ls -l)
+echo
+
+mkdir -p "$WATCHDOG_PID" "$WATCHDOG_FILE"
+touch "$WATCHDOG_PID/zedbox.pid" "$WATCHDOG_FILE/zedbox.touch"
+
+if [ -c $TPM_DEVICE_PATH ] && ! [ -f $DEVICE_KEY_NAME ]; then
+    # It is a device with TPM, enable disk encryption
+    if ! $BINDIR/vaultmgr setupDeprecatedVaults; then
+        echo "$(date -Ins -u) onboot.sh: vaultmgr setupDeprecatedVaults failed"
+    fi
+fi
+
+if [ -f $PERSISTDIR/reboot-reason ]; then
+    echo "Reboot reason: $(cat $PERSISTDIR/reboot-reason)" > /dev/console
+elif [ -n "$FIRSTBOOT" ]; then
+    echo "Reboot reason: NORMAL: First boot of device - at $(date -Ins -u)" > /dev/console
+else
+    echo "Reboot reason: UNKNOWN: reboot reason - power failure or crash - at $(date -Ins -u)" > /dev/console
+fi
```

Comment:
Do these things get logged somewhere?

Reply:
Yes, inside the console (at least in qemu VM) I can see them as before.

Reply:
But I don't know if stdout is captured and appears in memlogd.
Can you try with running a 
/usr/bin/logread -F -socket /run/memlogdq.sock | grep 'Reboot reason'

If it appears there then it will be sent to the controller log API.

Reply:
You are right as usual, thank you. Seems we had this behavior before as well, but nice to print this line through our logs. I modified the code and now can see:
```
8bc8dded-a337-4f05-9bd8-693d7a675959:~# logread |grep 'Reboot reason'
2022-11-21T08:07:32Z,onboot.006-pillar-onboot.out;Reboot reason: NORMAL: First boot of device - at 2022-11-21T08:07:32,858119713+00:00
```

Reply:
But does that line get picked up and reformatted by loguploader? That involves json encoding it etc, and it should be placed in /persist/newlog/ somewhere and sent to the controller in a format similar to the current log line.

Reply:
I can see the line in Adam's logs:
```
$ ./eden log --format=json|grep 'Reboot reason'
{"severity":"info","source":"onboot.006-pillar-onboot.out","content":"Reboot reason: NORMAL: First boot of device - at 2022-11-22T16:02:37,447877837+00:00\n","msgid":"98","timestamp":"2022-11-22T16:02:37Z"}
```

Reply:
And how does the same information look like in Adam's log prior to these changes?
I'm trying to make sure things in e.g., Kibana can still find it, and I think the logs from pillar have a msg field and not a content field.

Reply:
Output from device-steps is available in such format:
`{"severity":"info","source":"pillar.out","content":"Starting device-steps\n","msgid":"486","timestamp":"2022-11-17T14:31:39Z"}`

But the line you mentioned in the thread was not available in logs as it was [redirected](https://github.com/lf-edge/eve/blob/a5757bcce13ae02215477a3ef8056243ea36ce59/pkg/pillar/scripts/device-steps.sh#L188-L194) to /dev/console. We can see only information from nodeagent/baseosmgr.

Reply:
Ah - I picked a bad line.

How about the line with "msg:Used percentage of /persist: "?
Currently I see that appearing in the controller/Kibana in the form
msg:Used percentage of /persist: 7 eveVersion:8.5.4-xen-amd64 file: func: image:IMGA msgid:1,093 pid:0 severity:info source:pillar.out

I assume the source fill be different in the onboot case, but will it have a "msg" or a "content" field?

Reply:
The source difference is fine; just wanted to make sure that the string has the same tag (and it is "content" in both cases).